Zvariant: Value should implement From<Arc<str>>

# Description
The Value struct in zvariant does not implement `From<Arc<str>>` even though the Str helper struct does.
This means you can no longer call `.into()` to convert your Arc into a Value like you can with &str or String.
More importantly though you now need to manually map the values  of Vecs or HashMaps that use `Arc<str>` to construct the Array or Dict variant.

# Example
```rust
use std::sync::Arc;
use zvariant::Value;

fn main() {
    let arced: Arc<str> = "Arc'ed &str".into();

    let v1: Value = "Value".into(); // This works because Value implements From<&str>
    let v2: Value = arced.into(); // This does not, even though Str implements From<Arc<str>>
    let v3 = Value::Str(arced.into()); // We can manually turn our Arc into the helper Str struct
    
    // But for Arrays or Dicts this does not work:
    let vec: Vec<Arc<str>> = ["Hello", "World", "!"].iter().map(|v| (*v).into()).collect();

    // This does not work because Array only implements From<Vec<T>> where T implements Into<Value<'a>>
    let v4 = Value::Array(vec.into());

    // Like above we need to manually convert the Arcs for this to work
    let mapped: Vec<Value> = vec
        .iter()
        .map(|v| Value::Str(Arc::clone(v).into()))
        .collect();
    let v5 = Value::Array(mapped.into());

}
```

# Compiler output
```
error[E0277]: the trait bound `zvariant::Structure<'_>: From<Arc<str>>` is not satisfied
 --> src/main.rs:9:27
  |
9 |     let v2: Value = arced.into(); // This does not, even though Str implements From<Arc<str>>
  |                           ^^^^ the trait `From<Arc<str>>` is not implemented for `zvariant::Structure<'_>`
  |
  = help: the following other types implement trait `From<T>`:
            `zvariant::Structure<'_>` implements `From<(T0, T1)>`
            `zvariant::Structure<'_>` implements `From<(T0, T1, T2)>`
            `zvariant::Structure<'_>` implements `From<(T0, T1, T2, T3)>`
            `zvariant::Structure<'_>` implements `From<(T0, T1, T2, T3, T4)>`
            `zvariant::Structure<'_>` implements `From<(T0, T1, T2, T3, T4, T5)>`
            `zvariant::Structure<'_>` implements `From<(T0, T1, T2, T3, T4, T5, T6)>`
            `zvariant::Structure<'_>` implements `From<(T0, T1, T2, T3, T4, T5, T6, T7)>`
            `zvariant::Structure<'_>` implements `From<(T0, T1, T2, T3, T4, T5, T6, T7, T8)>`
          and 8 others
  = note: required for `Arc<str>` to implement `Into<zvariant::Structure<'_>>`
  = note: required for `zvariant::Value<'_>` to implement `From<Arc<str>>`
  = note: 1 redundant requirement hidden
  = note: required for `Arc<str>` to implement `Into<zvariant::Value<'_>>`

error[E0277]: the trait bound `zvariant::Structure<'_>: From<Arc<str>>` is not satisfied
  --> src/main.rs:16:31
   |
16 |     let v4 = Value::Array(vec.into());
   |                               ^^^^ the trait `From<Arc<str>>` is not implemented for `zvariant::Structure<'_>`
   |
   = help: the following other types implement trait `From<T>`:
             `zvariant::Structure<'_>` implements `From<(T0, T1)>`
             `zvariant::Structure<'_>` implements `From<(T0, T1, T2)>`
             `zvariant::Structure<'_>` implements `From<(T0, T1, T2, T3)>`
             `zvariant::Structure<'_>` implements `From<(T0, T1, T2, T3, T4)>`
             `zvariant::Structure<'_>` implements `From<(T0, T1, T2, T3, T4, T5)>`
             `zvariant::Structure<'_>` implements `From<(T0, T1, T2, T3, T4, T5, T6)>`
             `zvariant::Structure<'_>` implements `From<(T0, T1, T2, T3, T4, T5, T6, T7)>`
             `zvariant::Structure<'_>` implements `From<(T0, T1, T2, T3, T4, T5, T6, T7, T8)>`
           and 8 others
   = note: required for `Arc<str>` to implement `Into<zvariant::Structure<'_>>`
   = note: required for `zvariant::Value<'_>` to implement `From<Arc<str>>`
   = note: 3 redundant requirements hidden
   = note: required for `Vec<Arc<str>>` to implement `Into<zvariant::Array<'_>>`
```

# Version details
rustc: 1.84.1 (e71f9a9a9 2025-01-27)
zvariant: 5.2.0
edition: 2021

--- a/zvariant/src/into_value.rs
+++ b/zvariant/src/into_value.rs
@@ -1,4 +1,4 @@
-use std::{collections::HashMap, hash::BuildHasher};
+use std::{borrow::Cow, collections::HashMap, hash::BuildHasher, sync::Arc};
 
 #[cfg(feature = "gvariant")]
 use crate::Maybe;
@@ -48,6 +48,9 @@ into_value_from_both!(u64, U64);
 into_value_from_both!(i64, I64);
 into_value_from_both!(f32, F64);
 into_value_from_both!(f64, F64);
+
+into_value!(Arc<str>, Str);
+into_value!(Cow<'a, str>, Str);
 
 into_value_from_both!(&'a str, Str);
 into_value_from_both!(Str<'a>, Str);

--- a/zvariant/src/into_value.rs
+++ b/zvariant/src/into_value.rs
@@ -51,6 +51,7 @@ into_value_from_both!(f64, F64);
 
 into_value!(Arc<str>, Str);
 into_value!(Cow<'a, str>, Str);
+into_value_from_both!(String, Str);
 
 into_value_from_both!(&'a str, Str);
 into_value_from_both!(Str<'a>, Str);
@@ -80,12 +81,6 @@ try_into_value_from_ref!(Maybe<'a>, Maybe);
 into_value!(Fd<'a>, Fd);
 #[cfg(unix)]
 try_into_value_from_ref!(Fd<'a>, Fd);
-
-impl From<String> for Value<'_> {
-    fn from(v: String) -> Self {
-        Value::Str(crate::Str::from(v))
-    }
-}
 
 impl<'v, 's: 'v, T> From<T> for Value<'v>
 where
@@ -133,12 +128,6 @@ where
 {
     fn from(value: HashMap<K, V, H>) -> Self {
         Self::Dict(value.into())
-    }
-}
-
-impl<'v> From<&'v String> for Value<'v> {
-    fn from(v: &'v String) -> Value<'v> {
-        Value::Str(v.into())
     }
 }
 

